### DISCUSSION TITLE
Good Vec2 implementation

### DISCUSSION THREAD

**Discussion by AnuraDev** [jrouwe/JoltPhysics#1421] [General]
So it's more like a question/place to get suggestions. As i can see you can't really benefit from SIMD in case of Vec2 the same way you do with Vec3(4) without wasting extra cycles (copying/preparing data) or memory (having Vec4 effectively) or i am wrong here (i tend to think what for Vec2 scalar (plain float) solution is better)? or what are the options?

**Comment by jrouwe** [comment]
I really can't tell you what's faster, the best way to find out is to measure it yourself using e.g. [Superluminal](https://superluminal.eu/).

What you can do is to pack 2 Vec2's in 1 Vec4 and make your algorithm work on 2 things at once (e.g. collide 1 ray against 2 triangles or 2 rays against 1 triangle).
